--- src/architect/BUILD.bazel
+++ src/architect/BUILD.bazel
@@ -10,6 +10,7 @@ bzl_library(
 bzl_library(
     name = "ng_application",
     srcs = ["ng_application.bzl"],
+    visibility = ["//visibility:public"],
     deps = [
         ":utils",
         "@aspect_rules_js//js:defs",
@@ -27,6 +28,7 @@ bzl_library(
 bzl_library(
     name = "ng_test",
     srcs = ["ng_test.bzl"],
+    visibility = ["//visibility:public"],
     deps = [
         ":utils",
         "@aspect_rules_js//js:defs",
