diff --git a/gazelle_haskell_modules/BUILD.bazel b/gazelle_haskell_modules/BUILD.bazel
--- a/gazelle_haskell_modules/BUILD.bazel
+++ b/gazelle_haskell_modules/BUILD.bazel
@@ -1,4 +1,4 @@
-load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+load("@rules_go//go:def.bzl", "go_library", "go_test")
 
 go_library(
     name = "gazelle_haskell_modules",
@@ -10,13 +10,13 @@ go_library(
     importpath = "github.com/tweag/gazelle_haskell_modules/gazelle_haskell_modules",
     visibility = ["//visibility:public"],
     deps = [
-        "@bazel_gazelle//config:go_default_library",
-        "@bazel_gazelle//label:go_default_library",
-        "@bazel_gazelle//language:go_default_library",
-        "@bazel_gazelle//repo:go_default_library",
-        "@bazel_gazelle//resolve:go_default_library",
-        "@bazel_gazelle//rule:go_default_library",
-        "@com_github_bazelbuild_buildtools//build:go_default_library",
-        "@io_bazel_rules_go//go/tools/bazel:go_default_library",
+        "@gazelle//config:go_default_library",
+        "@gazelle//label:go_default_library",
+        "@gazelle//language:go_default_library",
+        "@gazelle//repo:go_default_library",
+        "@gazelle//resolve:go_default_library",
+        "@gazelle//rule:go_default_library",
+        "@gazelle_haskell_modules_buildtools_bridge//build:build",
+        "@rules_go//go/tools/bazel:go_default_library",
     ],
 )
 
