diff --git a/rs/extensions.bzl b/rs/extensions.bzl
--- a/rs/extensions.bzl
+++ b/rs/extensions.bzl
@@ -333,0 +334 @@ def _generate_hub_and_spokes(
+            extra_compile_data = annotation.extra_compile_data,
@@ -339,0 +341 @@ def _generate_hub_and_spokes(
+            rustc_env = annotation.rustc_env,
@@ -909,3 +912,3 @@ _annotation = tag_class(
-        # "compile_data": _relative_label_list(
-        # doc = "A list of labels to add to a crate's `rust_library::compile_data` attribute.",
-        # ),
+        "extra_compile_data": _relative_label_list(
+            doc = "A list of labels to add to a crate's generated `rust_library::compile_data` attribute.",
+        ),
@@ -974,3 +977,3 @@ _annotation = tag_class(
-        # "rustc_env": attr.string_dict(
-        #     doc = "Additional variables to set on a crate's `rust_library::rustc_env` attribute.",
-        # ),
+        "rustc_env": attr.string_dict(
+            doc = "Additional variables to set on a crate's generated `rust_library::rustc_env` attribute.",
+        ),
diff --git a/rs/private/repository_utils.bzl b/rs/private/repository_utils.bzl
--- a/rs/private/repository_utils.bzl
+++ b/rs/private/repository_utils.bzl
@@ -301,0 +302 @@ rust_crate_attrs = {
+    "extra_compile_data": attr.string_list(default = []),
@@ -305,0 +307 @@ rust_crate_attrs = {
+    "rustc_env": attr.string_dict(),
diff --git a/rs/private/annotations.bzl b/rs/private/annotations.bzl
--- a/rs/private/annotations.bzl
+++ b/rs/private/annotations.bzl
@@ -12,0 +13 @@ def _crate_annotation(
+        extra_compile_data = [],
@@ -18,0 +20 @@ def _crate_annotation(
+        rustc_env = {},
@@ -38,0 +41 @@ def _crate_annotation(
+        extra_compile_data = extra_compile_data,
@@ -44,0 +48 @@ def _crate_annotation(
+        rustc_env = rustc_env,
